Automattic/jetpack
# WordAds: Update EU Cookie Law Widget w/ ads compliance features

#### Changes proposed in this Pull Request:

* Add new setting for:
  * Banner consent expiration. 
_This updates the current default (30 days) to be a user-specified setting. New default is 180 days (6mo)._
* Add new filter `jetpack_disable_eu_cookie_law_widget`, which can be used to disable the banner.
  * Usage example: a site using the [MaxMind GeoIP database](https://www.maxmind.com/en/geoip-demo) on their server could write a filter that extrapolates each visitor's country from their IP and uses that to show banners for users from certain countries.
* Adds requirement (+ notice) that explicit button-click opt-in for ads module users.
* Adds a new `personalized-ads-consent` cookie for ads-enabled sites. 
The cookie banner will fire if either cookie is missing, and a consent action will update both cookies with the new date value. This method will cover use cases where users gain consent from users, but later turn on ads (which they may not have consent to personalize).

* Additional UI changes
  * Cherry-picked the changes from #9559 & #9463 which also update some language, so we can get a holistic review on this PR. Most notable is that the widget has been renamed for clarity.

  * Reordered the widget form sections in a more intuitive way. (Banner display > consent options > coloring)

  * Updated the default link text to be a part of the widget form. Previously, To find out more, including how to control cookies, see here: would be pre-pended to the widget link, but there was no way for the user to control this text. This seemed confusing to me when we had a textarea for that purpose.

  * Updated the widget output to allow users to use line breaks in the banner text field (just as we are in the default text).

  * Updated the widget form section labels to address some of the clarity issues you mentioned above.


<img width="315" alt="screen shot 2018-05-17 at 11 31 08 am" src="https://user-images.githubusercontent.com/2522431/40187575-d6bc54d2-59c5-11e8-9e55-d43674c91bf7.png">

#### Testing instructions:

* Test the new settings, make sure they behave as expected.
* Enable the Ads module, then click the consent banner. Make sure the `personalized-ads-consent` cookie fires.

#### Additional Notes:

* This will require updating https://jetpack.com/support/cookies/ to include ad cookie information (as well as the `personalized-ads-consent` cookie introduced here.
